include/alpaka/vec/Vec.hpp

```diff
@@ -511,8 +511,7 @@ namespace alpaka
         }
 
     private:
-        // 16 Byte alignment for usage inside of CUDA kernels.
-        alignas(16u) TSize m_data[TDim::value];
+        TSize m_data[TDim::value];
```

Comment:
The type `TSize` not always a primitive build in type.
If you placed the second version as variable in a kernel you waste the padded bytes as register.
Yes float4 in cuda is aligned but float3 is not but both are native types. This class is templated and the type is not specified.

Am 24. November 2015 20:22:38 MEZ, schrieb Benjamin Worpitz <notifications@github.com>:
>> @@ -511,8 +511,7 @@ namespace alpaka
>>          }
>>  
>>      private:
>> -        // 16 Byte alignment for usage inside of CUDA kernels.
>> -        alignas(16u) TSize m_data[TDim::value];
>> +        TSize m_data[TDim::value];
>
>The first version does not do anything and is not necessary because the
>natural alignment of a float is 4 bytes. Therefore the internal array
>of floats requires 4 byte alignment which then is the alignment
>requirement for the whole Vec.
>On CPU as well as on GPU all alignments other then 16 bytes for this
>type would prevent vectorization. The float4/vec4 data types of CUDA
>and OpenGL are all 16 byte aligned.
>There are only ever registers wasted when the Vec would be used within
>a struct when there is a member with smaller alignment before it.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/ComputationalRadiationPhysics/alpaka/pull/103/files#r45781409

-- 
Diese Nachricht wurde von meinem Android-Mobiltelefon mit K-9 Mail gesendet.